TestGroupByReverse flakiness

Spotted in CI:
```
--- FAIL: TestGroupByReverse (0.02s)
    handlers_test.go:156: 
        	Error Trace:	handlers_test.go:156
        	Error:      	Not equal: 
        	            	expected: "#0:record[ts:time,count:uint64]\n0:[2;1;]\n0:[1;2;]\n"
        	            	actual  : "#0:record[ts:time,count:uint64]\n0:[1;2;]\n0:[2;1;]\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 #0:record[ts:time,count:uint64]
        	            	+0:[1;2;]
        	            	 0:[2;1;]
        	            	-0:[1;2;]
        	            	 
        	Test:       	TestGroupByReverse
FAIL
```

--- a/driver/compile.go
+++ b/driver/compile.go
@@ -9,6 +9,7 @@ import (
 	"github.com/brimsec/zq/compiler"
 	"github.com/brimsec/zq/compiler/ast"
 	"github.com/brimsec/zq/compiler/kernel"
+	"github.com/brimsec/zq/field"
 	"github.com/brimsec/zq/pkg/nano"
 	"github.com/brimsec/zq/ppl/zqd/worker"
 	"github.com/brimsec/zq/proc"
@@ -53,7 +54,11 @@ func compile(ctx context.Context, program ast.Proc, zctx *resolver.Context, read
 	if cfg.Span.Dur == 0 {
 		cfg.Span = nano.MaxSpan
 	}
-	runtime, err := compiler.New(zctx, program)
+	var sortKey field.Static
+	if cfg.ReaderSortKey != "" {
+		sortKey = field.New(cfg.ReaderSortKey)
+	}
+	runtime, err := compiler.NewWithSortedInput(zctx, program, sortKey, cfg.ReaderSortReverse)
 	if err != nil {
 		return nil, err
 	}

